app/assets/javascripts/miq_dynatree.js

```diff
@@ -569,3 +569,186 @@ function miqSquashToggle(treeName) {
     ManageIQ.tree.expandAll = true;
   }
 }
+
+function miqTreeEventSafeEval(func, arg1, arg2, arg3, arg4) {
```

Comment:
We could get rid of these hardcoded arguments by returning the function instead..

if you do `return miqOnCheckProtect;` instead of calling it, and use `miqTreeEventSafeEval` as `miqTreeEventSafeEval(options.onclick)(node.data.key)` instead of `miqTreeEventSafeEval(options.onclick, node.data.key)` it should get a bit more maintanable :).

And having done that, this function can be rewritten as something like:
```js
var whitelist = ['miqOnCheckProtect', ...];
if (whitelist.includes(func)) {
  return window[func];
} else {
  throw new Error("Function not in whitelist: " + func);
}
```